egs/wsj/s5/steps/cleanup/do_cleanup_segmentation.sh

```diff
@@ -0,0 +1,147 @@
+min_correct_frames=0          # Minimum number of frames required in a subsegment to be kept
+max_segment_wer=20000         # Maximum WER% of an utterance, above which it is simply removed
+min_wer_for_splitting=-1      # Minimum WER% for a segment to be considered for splitting
+
+group_utt_words=              # If provided, build LMs on group of utterances 
+                              # that have at least these many words.  
+                              # Otherwise, build per-utterance LM
+
+# ngram options
+ngram_order=2                 
+interpolation_weights=0:0.1:0.9       # LM combination weights for 0-gram:1-gram:2-gram
+top_words_interpolation_weight=0.1    # Interpolation weight for top-words unigram
+top_n_words=100                       # Number of top-words to use for building a unigram LM
+
+lmwt=10         
+
+cmd=run.pl
+nj=4
+
+. ./path.sh
+. utils/parse_options.sh 
+
+
+if [ $# -ne 6 ]; then
+  echo "Usage: $0 <data> <lang> <srcdir> <dir> <segmentation-dir> <out-data>"
+  echo " e.g.: $0 data/sdm1/train_sp data/lang exp/sdm1/tri4a_ali_train_sp exp/sdm1/tri4a_sdm1_train_sp_bad_utts exp/sdm1/tri4a_sdm1/train_sp_bad_utts/segmentation_cleaned data/sdm1/train_sp_cleaned"
+  exit 1
+fi
+
+data=$1
+lang=$2
+srcdir=$3
+dir=$4
+segmentation_dir=$5
+out_data=$6
+
+dataid=`basename $data`
+
+mkdir -p $dir
+cp $srcdir/final.mdl $dir
+cp $srcdir/tree $dir
+cp $srcdir/cmvn_opts $dir
+cp $srcdir/{splice_opts,final.mat,final.alimdl} $dir || true
+chmod +w $dir/{final.mdl,tree,cmvn_opts}
+chmod +w $dir/{splice_opts,final.mat,final.alimdl} || true
+
+if [ $stage -le 6 ]; then
+  if [ -z "$group_utt_words" ]; then
+    # For per-utterance LMs
+    utils/copy_data_dir.sh $data $dir/${dataid}_split
+    cp $dir/${dataid}_split/text $dir/${dataid}_split/text.orig
+    awk '{print $1" "$1}' $data/utt2spk > $dir/${dataid}_split/orig2utt
+  else
+    # Group utterances such that each group has at least $group_utt_words
+    utils/copy_data_dir.sh $data $dir/${dataid}_split
+    python steps/cleanup/group_utterances.py --words-threshold=$group_utt_words \
+      $data/text $dir/${dataid}_split/text.orig $dir/${dataid}_split/orig2utt
+  fi
+fi
+ 
+if [ $stage -le 2 ]; then
+  echo "$0: Building biased language model..." 1>&2
+  steps/cleanup/make_biased_lm_graph.sh \
+    --nj $nj --cmd "$cmd" \
+    --ngram-order $ngram_order \
+    --top-words-interpolation-weight $top_words_interpolation_weight \
+    --interpolation-weights "$interpolation_weights" \
+    $dir/${dataid}_split $lang $srcdir $dir/graphs_${dataid}
+fi
+
+if [ $stage -le 3 ]; then
+  echo "$0: Decoding with biased language models..." 1>&2
+  steps/cleanup/decode_segmentation.sh \
+    --nj $nj --cmd "$cmd --mem 4G" --skip-scoring true --allow-partial false \
+    $dir/graphs_${dataid} \
+    $dir/${dataid}_split $dir/lats
+fi
+
+if [ $stage -le 4 ]; then
+  echo "$0: Get hypotheses from the decoded lattices using $hypothesis_method..." 1>&2
+  steps/cleanup/analyze_train_lats.sh \
```

Comment:
I don't really like the name 'analyze_train_lats.sh' because it's very vague about what analysis is being done.  How about something like lattice_oracle_align.sh, if that's what it's doing?